src/io/fileread.h

```diff
@@ -34,25 +34,30 @@
 class FileRead : public StreamRead {
 public:
 	struct Pos {
-		Pos(size_t const p = 0) : pos(p) {
+		Pos(size_t const p = 0) : pos(p) {  // NOLINT allow implicit conversion
 		}
 		/// Returns a special value indicating invalidity.
 		static Pos null() {
-			return std::numeric_limits<size_t>::max();
+			return Pos(std::numeric_limits<size_t>::max());
 		}
 
+		[[nodiscard]] bool operator==(const Pos& p) const {
+			return pos == p.pos;
+		}
 		[[nodiscard]] bool is_null() const {
 			return *this == null();
 		}
-		operator size_t() const {
+		operator size_t() const {  // NOLINT allow implicit conversion
 			return pos;
 		}
-		Pos operator++() {
-			return ++pos;
+		Pos& operator++() {
```

Comment:
not sure but shouldn't the ++ and += operator not get the [[nodiscard]] mark as well. same below in filewrite.h

Reply:
No because they modify the object directly.
```c++
x += y;
x++;
```
are perfectly valid statements. 

Such operators additionally return a reference to themselves so you can write chain statements like `x = y += ++z`. But the object is modified in-place so you don't usually need to use the return value.